AEPServices/Tests/services/NamedCollectionDataStoreTest.swift

```diff
@@ -385,6 +385,34 @@ class NamedCollectionDataStoreTest: XCTestCase {
         XCTAssertEqual(subscriptResult?.id, val2.id)
     }
 
+    func testSetDate() {
+        let date = Date()
+        store?.setObject(key: OBJ_KEY, value: date)
+
+        XCTAssertTrue(mockKeyValueService.setCalled)
+        if #available(iOS 13, tvOS 13, *) {
+            let encodedDate = try? JSONEncoder().encode(date)
+            XCTAssertEqual(mockKeyValueService.setValue as? Data, encodedDate)
+        } else {
+            XCTAssertEqual(mockKeyValueService.setValue as? Double, date.timeIntervalSince1970)
+        }
+    }
+    func testGetDateCodable() {
```

Comment:
can we also add a test where we mock the store as double and read as date?

Reply:
The next test case does it.